doc/source/serve/doc_code/hpu_inference_serve.py

```diff
@@ -0,0 +1,154 @@
+                break
+            except Empty:
+                await asyncio.sleep(0.001)
+
+    # generate the repsonse given input in a streaming manner
+    def streaming_generate(self, prompt, streamer, **config):
+        input_ids = self.tokenize(prompt)
+        self.model.generate(input_ids, streamer=streamer, **config)
+
+    # handle requests
+    async def __call__(self, http_request):
+        json_request: str = await http_request.json()
+        prompts = []
+        text = json_request["text"]
+        config = json_request["config"] if "config" in json_request else {}
+        streaming_response = json_request["stream"]
+        # prepare prompts
+        if isinstance(text, list):
+            prompts.extend(text)
+        else:
+            prompts.append(text)
+        # process config
+        if "max_new_tokens" not in config:
+            # hpu requires setting max_new_tokens
+            config["max_new_tokens"] = 128
+        # enable hpu graph runtime
+        config["hpu_graphs"] = True
+        # lazy mode should be True when using hpu graphs
+        config["lazy_mode"] = True
+        # non streaming
+        if not streaming_response:
+            return self.generate(prompts, **config)
+        from starlette.responses import StreamingResponse
+        from transformers import TextIteratorStreamer
+        from functools import partial
+
+        streamer = TextIteratorStreamer(
+            self.tokenizer, skip_prompt=True, timeout=0, skip_special_tokens=True
+        )
+        self.loop.run_in_executor(
+            None, partial(self.streaming_generate, prompts, streamer, **config)
+        )
+        return StreamingResponse(
+            self.consume_streamer_async(streamer),
+            status_code=200,
+            media_type="text/plain",
+        )
+
+
+# __model_def_end__
+
+# __main_code_start__
+if __name__ == "__main__":
+    import sys
+    import ray
+    import requests
+
+    ray.init(address="auto")
+    if len(sys.argv) > 1:
+        model_id_or_path = sys.argv[1]
+    else:
+        model_id_or_path = "meta-llama/Llama-2-7b-chat-hf"
+    deployment = LlamaModel.bind(model_id_or_path)
+    handle = serve.run(deployment, _blocking=True)
+    print("Model is deployed successfully at http://127.0.0.1:8000/")
+    input("Press Enter to start inference")
+    prompt = "Once upon a time,"
+    # Add generation config here
+    config = {}
+    sample_input = {"text": prompt, "config": config, "stream": False}
+    outputs = requests.post("http://127.0.0.1:8000/", json=sample_input, stream=False)
+    print(outputs.text, flush=True)
+    sample_input["stream"] = True
+    input("Press Enter to start streaming inference")
+    outputs = requests.post("http://127.0.0.1:8000/", json=sample_input, stream=True)
+    outputs.raise_for_status()
+    for output in outputs.iter_content(chunk_size=None, decode_unicode=True):
+        print(output, end="", flush=True)
+    print()
+
+# __main_code_end__
```

Comment:
Could you split the code example, so the entire Serve app is stored on one file, and the client script to query it is on another file? Ideally, the user should be able to run the serve app by running:

```
% ray start --head
% serve run your_example:app
```

and then separately query it using

```
python client.py
```

This makes it simple to productionize the example since the user can build a Serve config file directly from the example.